### PR TITLE
[BCI-3339] Change signature of `SubmitSignedTransaction` in `ChainWriter`

### PR DESCRIPTION
**Pull request by nickcorin**
core ref: bff2850313ca3916cde85716956196d5ab620944

### CODE REVIEW

File: pkg/types/chain_writer.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ type ChainWriter interface {
 	// SubmitSignedTransaction packs and broadcasts a transaction to the underlying chain.
 	//
 	// The `transactionID` will be used by the underlying TXM as an idempotency key, and unique reference to track transaction attempts.
-	SubmitSignedTransaction(ctx context.Context, payload []byte, signature map[string]any, transactionID uuid.UUID, toAddress string, meta *TxMeta, value big.Int) error
+	SubmitSignedTransaction(ctx context.Context, args []any, transactionID uuid.UUID, toAddress string, meta *TxMeta, value big.Int) error
```
Comment (jmank88) [inline pkg/types/chain_writer.go:14]: Can we update the comment doc? `args []any` is not as self-descriptive

**Review by silaslenihan — DISMISSED** [review]

**Review by amit-momin — DISMISSED** [review]

File: pkg/types/chain_writer.go (hunk 1 above)
Comment (archseer) [inline pkg/types/chain_writer.go:14]: Let's also rename this to `CreateTransaction`

**Review by archseer — CHANGES_REQUESTED** [review]

File: pkg/types/chain_writer.go [hunk 2]
```diff
@@ -8,10 +8,11 @@ import (
 )
 
 type ChainWriter interface {
-	// SubmitSignedTransaction packs and broadcasts a transaction to the underlying chain.
+	// SubmitTransaction packs and broadcasts a transaction to the underlying chain.
 	//
-	// The `transactionID` will be used by the underlying TXM as an idempotency key, and unique reference to track transaction attempts.
-	SubmitSignedTransaction(ctx context.Context, payload []byte, signature map[string]any, transactionID uuid.UUID, toAddress string, meta *TxMeta, value big.Int) error
+	// - `args` specifies input parameters to the contract call.
```
Comment (jmank88) [inline pkg/types/chain_writer.go:13]: What are the limitations on the types that can be passed as `args`? b/c we cannot literally support `any`
We need to say something precise, just like `ChainReader` does:
https://github.com/smartcontractkit/chainlink-common/blob/main/pkg/types/chain_reader.go#L25-L29

Comment (jmank88) [reply pkg/types/chain_writer.go:13]: If there is sufficient overlap between the expectations here (even just for serialization), then we may want to declare a new type (alias) that we can document just once, and then use in any related methods, without duplicating the same sort of information around.

@nolag WDYT?

File: pkg/types/chain_writer.go (hunk 1 above)
Comment (nickcorin) [reply pkg/types/chain_writer.go]: Mh. I'm not sure about the rename. We're wrapping `CreateTransaction`, but this method only prepares and submits the transaction to the relevant TXM.

File: pkg/types/chain_writer.go (hunk 2 above)
Comment (nickcorin) [reply pkg/types/chain_writer.go:13]: Yeah agreed. These args are likely always going to be the arguments passed into the tx to trigger some contract function.

One constraint we have, however, is that the types need to be serializable in some way to be passed over the wire.

**Review by archseer — APPROVED** [review]
Merging this to unblock changes in core -- typing improvements can be a follow-up

**Review by bolekk — APPROVED** [review]

**Pull request merged**